FileWriteFlags.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace FileWriteFlag
+{
+	// Default values are Do not overwrite files, append to existing files, and write in binary format.
+	enum FileWriteFlag : unsigned int
+	{
+		// SET:     Allow overwritting the file if it exists (mutually exclusive from Append).
+		// NOT SET: Do not allow overwritting an exisitng file.
+		Overwrite = 0b0000'0001,
+
+		// SET:     Do not allow a new file to be created (Mutually exclusive from Overwrite).
+		// NOT SET: Allow new file creation. Overwrite must also be set.
+		DoNotCreate = 0b0000'0010,
+
+		// SET:     If file exists, default to truncating file to zero bytes
+		// NOT SET: If file exists, default to writing at end of file (Append).
```

Comment:
This doesn't seem like the opposite of Append, particularly for read streams, where you'd almost certainly not want to truncate, nor append. For a write stream, you might want to open an existing file, preserving the contents, yet still start to overwrite from the beginning.